src/Nixfmt/Parser.hs

```diff
@@ -196,6 +196,7 @@ indentedStringPart =
       ( chunk "''\\n"
           <|> chunk "''\\r"
           <|> chunk "''\\t"
+          <|> chunk "''\\'"
           <|> chunk "''\\"
           *> (Text.singleton <$> anySingle)
             <|> chunk "''$"
```

Comment:
Ohh, I'm now seeing the problem: The formatting here is very misleading, because it makes me believe that `<|>` binds stronger than `*>`, but in reality it's the other way around: `<|>` is [`infixl 3`](https://hackage.haskell.org/package/base-4.20.0.1/docs/Control-Applicative.html#v:-60--124--62-), while `*>` is [`infixl 4`](https://hackage.haskell.org/package/base-4.20.0.1/docs/Control-Applicative.html#v:-42--62-).

It makes more sense when we format it this way, also added a comment:
```haskell
chunk "''\\n"
<|> chunk "''\\r"
<|> chunk "''\\t"
<|> chunk "''\\'"
-- ''\ followed by any char throws away the ''\ part, leaving just the char
<|> chunk "''\\" *> (Text.singleton <$> anySingle)
<|> chunk "''$"
<|> chunk "'''"
<|> chunk "$$"
<|> try (chunk "$" <* notFollowedBy (char '{'))
<|> try (chunk "'" <* notFollowedBy (char '\''))
<|> someP (\t -> t /= '\'' && t /= '$' && t /= '\n')
```